src/main/java/com/mongodb/jdbc/logging/LoggingAspect.aj

```diff
@@ -0,0 +1,73 @@
+package com.mongodb.jdbc.logging;
+
+import com.mongodb.MongoException;
+import java.sql.SQLException;
+import java.util.Locale;
+import java.util.Properties;
+import java.util.logging.Level;
+import org.aspectj.lang.JoinPoint;
+import org.bson.BsonValue;
+
+public aspect LoggingAspect perthis(execution(com.mongodb.jdbc.*.new(..)))
+{
+    private MongoLogger logger = null;
+
+    pointcut setLogger(MongoLogger logger): set(MongoLogger *.*) && args(logger) && !within(LoggingAspect) ;
```

Comment:
Nice on the `!within`. Accidentally instrumenting the instrumentation code is a common footgun that bites people. Never want to see infinite instrumentation overflows ;)